src/allmydata/storage/http_server.py

```diff
@@ -530,6 +530,60 @@ def _cddl_validation_error(_, request, failure):
         return str(failure.value).encode("utf-8")
 
 
+async def read_encoded(
+    reactor, request, schema: Schema, max_size: int = 1024 * 1024
+) -> Any:
+    """
+    Read encoded request body data, decoding it with CBOR by default.
+
+    Somewhat arbitrarily, limit body size to 1MiB by default.
+    """
+    content_type = get_content_type(request.requestHeaders)
+    if content_type is None:
```

Comment:
This and the next line are new, otherwise this function is unchanged from previous location.